### PR TITLE
CodeJam-3 CoreJS

### PR DESCRIPTION
**Pull request by DrDiman**
I've done 2 functions, mocha tests and package.json with syle-guide airbnb/base, but I've not implemented "recursion" function fully... I've tried...

### CODE REVIEW

File: make.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = function make(...arg) {
+  const myArr = [...arg];
```
Comment (den-bozhyk) [inline make.js]: why do you need spread here? it is redundant
this construction is equal to 
const myArr = arg;

File: make.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  function myFunc(current) {
+    if (typeof current === 'function') {
+      return myArr.reduce(current);
+    }
+    for (let i = 0; i < arguments.length; i = 1 + i) {
```
Comment (den-bozhyk) [inline make.js]: is there any reason why you don't like i++?
anyway it could be simplified
from i = 1 + i 
to i += 1

File: sumOfOther.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module.exports = function sumOfOther(arr) {
+  return arr.map(el => arr.reduce((a, b) => a + b) - el);
```
Comment (den-bozhyk) [inline sumOfOther.js]: FYI:
this is O(n^2) time complexity and it could be optimized to O(n)

File: recursion.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+module.exports = function recursion(arg) {
```
Comment (den-bozhyk) [inline recursion.js]: doesn't look like correct algorithm, did you check it with example input from the description?
as I can see, tests are not passed for this task

File: recursion.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    const val = objTree.value;
+    arrayTree.push([val]);
+
+    if (!(objTree.hasOwnProperty('left')) || !(objTree.hasOwnProperty('right'))) {
+      return arrayTree;
+    }
+
+    if (objTree.hasOwnProperty('left')) {
+      objTree = objTree.left;
+      return myfunc(objTree);
```
Comment (den-bozhyk) [inline recursion.js]: you are returning from here, so it means you will never traverse right part of the tree

File: package.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/DrDiman/rsschool-codejam/issues"
+  },
+  "homepage": "https://github.com/DrDiman/rsschool-codejam#readme",
+  "keywords": [],
+  "dependencies": {
+    "eslint": "^5.8.0",
```
Comment (den-bozhyk) [inline package.json]: eslint should be in dev dependencies

File: make.js (hunk 2 above)
Comment (DrDiman) [reply make.js]: i used it due to airbnb-style) I usually use i++

File: recursion.js (hunk 4 above)
Comment (DrDiman) [reply recursion.js]: the only i've done were tests for that function...  I didn't implement it =(

Comment (den-bozhyk) [reply recursion.js]: why you didn't implement it? and where did you get implementation?

Comment (DrDiman) [reply recursion.js]: I did it, but it didn't work correctly) 
I typed only my own tests for the function... and didn't find out how to create correct recursion...

**Pull request closed without merging**